### ISSUE TITLE
Error when trying to add an item to the custom list

### ISSUE THREAD

**Issue by Scoboose** [lgallucci/LoonBestInSlot#16]
Edit:  SOD version

LoonBestInSlot/UI/AutoComplete.lua:42: attempt to index local 'button' (a nil value)
[string "@LoonBestInSlot/UI/AutoComplete.lua"]:42: in function <LoonBestInSlot/UI/AutoComplete.lua:40>
[string "@LoonBestInSlot/UI/AutoComplete.lua"]:184: in function `action'
[string "@LoonBestInSlot/UI/AutoComplete.lua"]:189: in function <LoonBestInSlot/UI/AutoComplete.lua:168>

Locals:
button = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index local 'button' (a nil value)"

**Comment by lgallucci** [comment]
Is this in WOTLK or in SOD ?

**Comment by Scoboose** [comment]
> Is this in WOTLK or in SOD ?

SOD

It happens after pressing enter when searching for an item

**Comment by lgallucci** [comment]
I'll look into it tonight, raiding atm ;)

**Comment by Scoboose** [comment]
No rush, thank you for the amazingly quick reply!

**Comment by lgallucci** [comment]
Is this happening when pressing enter immediately?  Are you copying and pasting in an item and hitting enter before the dropdown populates?  I seem to have replicated and fixed that, but I want to make sure.  See if waiting a sec before hitting enter doesn't cause the error.

**Comment by Scoboose** [comment]
Sorry I was unable to test yesterday. When I type in an item and wait it does have a drop down and I can select the item.  In doing these tests I realized I was also searching for some items that the addon does not find. For example, https://www.wowhead.com/classic/item=14252/lunar-coronet and https://www.wowhead.com/classic/item=10502/spellpower-goggles-xtreme and https://www.wowhead.com/classic/item=217298/mantle-of-doan

**Comment by lgallucci** [comment]
So.  I was only previously populating the autocomplete with items already in the addon ( i.e. in one of the lists already ) which worked ok in wotlk b/c most of the bis items were in raids and/or dungeons that are already known.  It seems to be less fine in sod.  Working on getting all the items to appear in the autocomplete, but i'm working around a current blizzard ui bug.  In the meantime i added all the dungeon & profession items up to uldaman.

**Comment by Scoboose** [comment]
Awesome, will you upload to curse?  What bug are you having? I had an issue with an addon I was making with drop downs.

**Comment by lgallucci** [comment]
I migrated to item mixins a while back to get item data from the servers, and there's a bug in the blizzard code in the item mixin for a bunch of items that no longer return data, but still show as true under C_Item.DoesItemExistByID.  

So there's no real way to know which item ids those are that don't return data and when you try to get the data for these it throws a blizzard lua error.  The only way I can figure out to fix it is to revert partially to the old item code.  

I've uploaded to curse, but it's been 'Under Review' for hours now.  They might be having problems.
